tests/providers/cncf/kubernetes/operators/test_custom_object_launcher.py

```diff
@@ -14,3 +14,125 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from unittest.mock import patch
+
+import pytest
+
+from airflow.exceptions import AirflowException
+from airflow.providers.cncf.kubernetes.operators.custom_object_launcher import (
+    SparkJobSpec,
+    SparkResources,
+)
+
+
+class TestSparkJobSpec:
+    @patch("airflow.providers.cncf.kubernetes.operators.custom_object_launcher.SparkJobSpec.update_resources")
+    @patch("airflow.providers.cncf.kubernetes.operators.custom_object_launcher.SparkJobSpec.validate")
+    def test_spark_job_spec_initialization(self, mock_validate, mock_update_resources):
+        entries = {
+            "spec": {
+                "dynamicAllocation": {
+                    "enabled": True,
+                    "initialExecutors": 1,
+                    "minExecutors": 1,
+                    "maxExecutors": 2,
+                },
+                "driver": {},
+                "executor": {},
+            }
+        }
+        SparkJobSpec(**entries)
+        mock_validate.assert_called_once()
+        mock_update_resources.assert_called_once()
+
+    def test_spark_job_spec_dynamicAllocation_enabled(self):
+        entries = {
+            "spec": {
+                "dynamicAllocation": {
+                    "enabled": True,
+                    "initialExecutors": 1,
+                    "minExecutors": 1,
+                    "maxExecutors": 2,
+                },
+                "driver": {},
+                "executor": {},
+            }
+        }
+        spark_job_spec = SparkJobSpec(**entries)
+
+        assert spark_job_spec.spec["dynamicAllocation"]["enabled"]
+
+    def test_spark_job_spec_dynamicAllocation_enabled_with_invalid_config(self):
+        entries = {
+            "spec": {
+                "dynamicAllocation": {
+                    "enabled": True,
+                    "initialExecutors": 1,
+                    "minExecutors": 1,
+                    "maxExecutors": 2,
+                },
+                "driver": {},
+                "executor": {},
+            }
+        }
+        with pytest.raises(
+            AirflowException, match="Make sure initial/min/max value for dynamic allocation is passed"
+        ):
+            cloned_entries = entries.copy()
+            cloned_entries["spec"]["dynamicAllocation"]["initialExecutors"] = None
+            SparkJobSpec(**cloned_entries)
+
+        with pytest.raises(
+            AirflowException, match="Make sure initial/min/max value for dynamic allocation is passed"
+        ):
+            cloned_entries = entries.copy()
+            cloned_entries["spec"]["dynamicAllocation"]["minExecutors"] = None
```

Comment:
```suggestion
        cloned_entries = entries.copy()
        cloned_entries["spec"]["dynamicAllocation"]["minExecutors"] = None
        with pytest.raises(
            AirflowException, match="Make sure initial/min/max value for dynamic allocation is passed"
        ):
```

Reply:
More context about why it is better move everything not related to the exception https://github.com/apache/airflow/pull/38219

In particular this case it is pretty safe to have it inside of `pytest.raises` because it also validate exception text on `match`. However if rule PT012 will add into the main check, that it will require to change this test anyway